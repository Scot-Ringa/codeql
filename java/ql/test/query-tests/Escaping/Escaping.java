@ThreadSafe
public class Escaping {
  int x;  //$ Alert
  public int y = 0;  //$ Alert
  private int z = 3;
  final int w = 0;
  public final int u = 4;
  private final long a = 5;
  protected long b = 0;  //$ Alert
  protected final long c = 0L;
  volatile long d = 3;
  protected volatile long e = 3L;

  public void methodLocal() {
    int i;
  }
}