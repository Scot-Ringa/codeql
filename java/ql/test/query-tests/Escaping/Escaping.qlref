query: Likely Bugs/Concurrency/Escaping.ql
postprocess: utils/test/InlineExpectationsTestQuery.ql