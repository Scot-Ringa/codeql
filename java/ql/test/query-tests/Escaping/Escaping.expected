| Escaping.java:3:7:3:7 | x | The class $@ is marked as thread-safe, but this field is potentially escaping. | Escaping.java:2:14:2:21 | Escaping | Escaping |
| Escaping.java:4:14:4:14 | y | The class $@ is marked as thread-safe, but this field is potentially escaping. | Escaping.java:2:14:2:21 | Escaping | Escaping |
| Escaping.java:9:18:9:18 | b | The class $@ is marked as thread-safe, but this field is potentially escaping. | Escaping.java:2:14:2:21 | Escaping | Escaping |
